sys/net/network_layer/ng_ipv6/ng_ipv6.c

```diff
@@ -308,7 +308,8 @@ static int _fill_ipv6_hdr(kernel_pid_t iface, ng_pktsnip_t *ipv6,
 
         /* We deal with multiple interfaces here (multicast) => possible
          * different source addresses => duplication of payload needed */
-        while (ptr != payload) {
+        while (ptr != payload->next) {
```

Comment:
I don't understand why you check for `payload->next` here instead of `payload`. Is `payload->next` the upper layer or lower layer? I seem to confuse this all the time..